src/EditorFeatures/Core/Implementation/AutomaticCompletion/BraceCompletionSessionProvider.BraceCompletionSession.cs

```diff
@@ -364,6 +352,72 @@ private void MoveCaretToClosingPoint()
                 }
             }
 
+            private void ApplyBraceCompletionResult(BraceCompletionResult result)
+            {
+                // Apply sets of edits incrementally and update the buffer on each set applied.
```

Comment:
this is really unfortunate but I don't have a better solution here.  Open to ideas.

I did try just having the service specify the new open/close locations, but unfortunately the act of applying an edit triggers the check on the editor side for the tracking spans, and I cant change the tracking spans before applying the edit...